pkg/es/config/config.go

```diff
@@ -188,10 +202,65 @@ func (c *Configuration) GetTagDotReplacement() string {
 }
 
 // GetConfigs wraps the configs to feed to the ElasticSearch client init
-func (c *Configuration) GetConfigs() []elastic.ClientOptionFunc {
+func (c *Configuration) GetConfigs(logger *zap.Logger) []elastic.ClientOptionFunc {
 	options := make([]elastic.ClientOptionFunc, 3)
+	if c.TLS.Enabled {
+		tlsConfig, err := c.CreateTLSConfig(logger)
+		if err != nil {
+			return nil
+		}
+		httpClient := &http.Client{
+			Timeout: c.Timeout,
+			Transport: &http.Transport{
+				TLSClientConfig: tlsConfig,
+			},
+		}
+		options[0] = elastic.SetHttpClient(httpClient)
+		options[1] = elastic.SetURL(c.Servers...)
+		options[2] = elastic.SetSniff(c.Sniffer)
+		return options
+	}
```

Comment:
can we instead use `options = append(options, ...)` and have a single return? I don't like that this duplicates SetURL/SetSniff, no reason to fork the logic this way.